shadows/framework/src/main/java/org/robolectric/shadows/ImageUtil.java

```diff
@@ -54,44 +54,50 @@ public static boolean writeToStream(
 
     try {
       ImageWriter writer = null;
-      Iterator<ImageWriter> iter = ImageIO.getImageWritersByFormatName(getFormatName(format));
+      // JPEG image writer can't process alpha correctly, so we use png image writer fix this
+      // problem.
+      Iterator<ImageWriter> iter = ImageIO.getImageWritersByFormatName("png");
```

Comment:
This line will break tests that try to compress to JPEG explicitly:
```
bitmap.compress(Bitmap.CompressFormat.JPEG, 100, fileOutputStream);
```

The output will be PNG regardless.

Reply:
If we can ensure this method doesn't process invalid image data of `JPEG` type, we can use the origin method to select `ImageWriter` based on the format of bitmap.

The `JPEG` doesn't support alpha channel, at least the doc of `Bitmap.compress` from Android SDK says(https://developer.android.com/reference/android/graphics/Bitmap#compress(android.graphics.Bitmap.CompressFormat,%20int,%20java.io.OutputStream) ):

> e.g. JPEG only supports opaque pixels

So we can use `JPEGWriter` to compress valid `JPEG` image data. But the old method also use `JPEGWriter` to compress `WEBP*` image data, that support alpha channel, at least lossless `WEBP`(https://developers.google.com/speed/webp/docs/webp_lossless_alpha_study). Maybe we should keep to use `PNGWriter` for `WEBP*` format, and use `JPEGWriter` for `JPEG` only.

If you think the above processing method is fine, I will update with new commit, and add basic warning info when the writer think the data is invalid.